controllers/operatorconfigmap_controller.go

```diff
@@ -707,34 +723,86 @@ func (c *OperatorConfigMapReconciler) reconcileSubscription() error {
 	}
 
 	if desiredChannel != "" {
-		subscriptions := &opv1a1.SubscriptionList{}
-		err := c.list(
-			subscriptions,
+		subList := &opv1a1.SubscriptionList{}
+		if err := c.list(
+			subList,
+			client.MatchingFields{subPackageIndexName: "ocs-client-operator"},
 			client.InNamespace(c.OperatorNamespace),
-			client.MatchingLabels{subscriptionLabelKey: subscriptionLabelValue},
 			client.Limit(1),
-		)
-		if err != nil {
-			return fmt.Errorf("failed to list subscription for ocs-client-operator using labels: %v", err)
+		); err != nil {
+			return fmt.Errorf("failed to list subscriptions: %v", err)
 		}
 
-		if len(subscriptions.Items) == 1 {
-			clientSubscription := &subscriptions.Items[0]
-			if desiredChannel != clientSubscription.Spec.Channel {
-				clientSubscription.Spec.Channel = desiredChannel
-				// TODO: https://github.com/red-hat-storage/ocs-client-operator/issues/130
-				// there can be a possibility that platform is behind, even then updating the channel will only make subscription to be in upgrading state
-				// without any side effects for already running workloads. However, this will be a silent failure and need to be fixed via above TODO issue.
-				if err := c.update(clientSubscription); err != nil {
-					return fmt.Errorf("failed to update subscription channel to %v: %v", desiredChannel, err)
-				}
+		if len(subList.Items) != 1 {
+			return fmt.Errorf("failed to find ocs-client-operator subscription")
+		}
+		clientSubscription := &subList.Items[0]
+		if desiredChannel != clientSubscription.Spec.Channel {
+			clientSubscription.Spec.Channel = desiredChannel
+			// TODO: https://github.com/red-hat-storage/ocs-client-operator/issues/130
+			// there can be a possibility that platform is behind, even then updating the channel will only make subscription to be in upgrading state
+			// without any side effects for already running workloads. However, this will be a silent failure and need to be fixed via above TODO issue.
+			if err := c.update(clientSubscription); err != nil {
+				return fmt.Errorf("failed to update subscription channel to %v: %v", desiredChannel, err)
 			}
 		}
 	}
 
 	return nil
 }
 
+func (c *OperatorConfigMapReconciler) reconcileDependentsSubscription() error {
```

Comment:
Why do we complicate this by addressing all depended as a single thing?
Please separate into a reconcile call / func for each subscription. 

If the implementation is identical at this point in time have one `reconcileDependentSubscription` function and call it from `reconcocilePhases` for each dependent. 

This will allow for each dependent operator reconciliation to evolve in a decoupled way from all the others

Reply:
reconciling addons separately.